src/components/MapView/Layers/AnalysisLayer/index.tsx

```diff
@@ -46,10 +47,11 @@ function AnalysisLayer() {
     analysisData instanceof ExposedPopulationResult
       ? (analysisData.statistic as string)
       : 'data';
+  const boundaryId = getBoundaryLayerSingleton().id;
```

Comment:
If the active layer has a unique boundary, `getBoundaryLayerSingleton` should return that unique boundary.

Reply:
@wadhwamatic can you confirm that we want the analysis to use the "unique boundary" when mixing a raster and admin level data with a unique boundary?

Reply:
@ericboucher - yes. whichever boundary associated with the 'admin level data' layer should be used for the analysis. I created an issue for this a few weeks ago: https://github.com/WFP-VAM/prism-frontend/issues/341